pkg/controller/core/clusterqueue_controller_test.go

```diff
@@ -0,0 +1,120 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1alpha2"
+)
+
+func TestUpdateCqStatus(t *testing.T) {
+	testCases := map[string]struct {
+		cqConditions     []metav1.Condition
+		conditionStatus  metav1.ConditionStatus
+		reason, msg      string
+		wantCqConditions []metav1.Condition
+		wantUpdated      bool
+	}{
+		"empty conditions": {
+			cqConditions:    nil,
+			conditionStatus: metav1.ConditionFalse,
+			reason:          "FlavorNotFound",
+			msg:             "Can't admit new workloads; some flavors are not found",
+			wantCqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionFalse,
+				Reason:  "FlavorNotFound",
+				Message: "Can't admit new workloads; some flavors are not found",
+			}},
+			wantUpdated: true,
+		},
+		"same condition status": {
+			cqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionTrue,
+				Reason:  "Ready",
+				Message: "Can admit new workloads",
+			},
+			},
+			conditionStatus: metav1.ConditionTrue,
+			reason:          "Ready",
+			msg:             "Can admit new workloads",
+			wantCqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionTrue,
+				Reason:  "Ready",
+				Message: "Can admit new workloads",
+			}},
+			wantUpdated: false,
+		},
+		"same condition status with different reason and message": {
+			cqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionFalse,
+				Reason:  "FlavorNotFound",
+				Message: "Can't admit new workloads; Can't admit new workloads; some flavors are not found",
+			}},
+			conditionStatus: metav1.ConditionFalse,
+			reason:          "Terminating",
+			msg:             "Can't admit new workloads; clusterQueue is terminating",
+			wantCqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionFalse,
+				Reason:  "Terminating",
+				Message: "Can't admit new workloads; clusterQueue is terminating",
+			}},
+			wantUpdated: true,
+		},
+		"different condition status": {
+			cqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionFalse,
+				Reason:  "FlavorNotFound",
+				Message: "Can't admit new workloads; some flavors are not found",
+			}},
+			conditionStatus: metav1.ConditionTrue,
+			reason:          "Ready",
+			msg:             "Can admit new workloads",
+			wantCqConditions: []metav1.Condition{{
+				Type:    kueue.ClusterQueueActive,
+				Status:  metav1.ConditionTrue,
+				Reason:  "Ready",
+				Message: "Can admit new workloads",
+			}},
+			wantUpdated: true,
+		},
+	}
+
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			cqStatus := &kueue.ClusterQueueStatus{Conditions: tc.cqConditions}
+			actualUpdated := UpdateCqStatus(cqStatus, 0, 0, kueue.UsedResources{}, tc.conditionStatus, tc.reason, tc.msg)
```

Comment:
I'm generally prefer not to test such small functions that are not exported. You could try to test the entire `Reconcile` or maybe even `UpdateCqStatusIfChanged`

Reply:
SGTM
I would change this test to one using `UpdateCqStatusIfChanged`.

Reply:
Done.